### PR TITLE
[Core] Fix the NoOp for rich status

### LINKED ISSUE #2665: `spot queue -r` failed to restart spot controller but `start` works
4e227cef

Repro, all instances on AWS:
- Get a stopped spot controller first (spot launch + wait for autostop/manually stop it)

```
» sky spot queue -r
Fetching managed spot job statuses...
Managed spot jobs
Failed to query spot jobs: [sky.exceptions.ResourcesUnavailableError] Failed to acquire resources to restart the stopped cluster sky-spot-controller-8a3968f2 in us-east-1. Please retry again later.
To keep retrying until the cluster is up, use the `--retry-until-up` flag.
```

However, `sky start sky-spot-controller-8a3968f2` restarts the controller without issues. Tried manually restarting in console too without issues.

The above is reproducible after a few tries. It seems `spot queue -r` may have run into some issues and thought it failed to restart.

### PR DESCRIPTION
**Pull request by Michaelvll**
Fixes #2665

Previously, when `NoOp` is returned, `with rich_utils.safe_status() as status:`, the `status` is actually the return value of `__enter__` (None for the `NoOp`), causing failure if `status` is used later.

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
  - [x] reproducible code in #2665.
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/utils/rich_utils.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ class _NoOpConsoleStatus:
     """An empty class for multi-threaded console.status."""
 
     def __enter__(self):
-        pass
+        return self
```
Comment (concretevitamin) [inline sky/utils/rich_utils.py:18]: Q: which call path into this caused the original issue?

**Review by concretevitamin — APPROVED** [review]
Thanks @Michaelvll, LGTM.

File: sky/utils/rich_utils.py (hunk 1 above)
Comment (Michaelvll) [reply sky/utils/rich_utils.py:18]: The ones like this: https://github.com/skypilot-org/skypilot/blob/f2cf0329ed7624624bf89d49637d498664cbbbe8/sky/provision/provisioner.py#L352-L353

The `status` will be None, when the provisioner is called with `sky_logging.is_silent()`, which is used for `sky spot queue`: https://github.com/skypilot-org/skypilot/blob/f2cf0329ed7624624bf89d49637d498664cbbbe8/sky/cli.py#L1620-L1623

Comment (concretevitamin) [reply sky/utils/rich_utils.py:18]: SG

**Pull request merged**